### PR TITLE
Kl dev

### PR DESCRIPTION
**Pull request by JonasKlamroth**
minor improvements to the ui:
- insert missing case statements now implemented
- rules applicable with default paramenters now get applied without confirmation dialog
- shift-click opens parameter-dialog even if not "necessary"
- rules cannot be applied to closed branches anymore

### CODE REVIEW

File: modules/fxgui/src/edu/kit/iti/algover/rule/script/ScriptController.java [hunk 1]
```diff
@@ -334,14 +328,124 @@ public void runScript() {
...
+     *
+     * @param pn the proofnode for which the cases should be created
+     * @param cases the cases that already exist
+     * @return a case statement containing all necesarry cases
+     */
+    private Statement createCasesForNode(ProofNode pn, List<CaseStatement> cases) {
+        CasesStatement res = new CasesStatement();
+        for(ProofNode p : pn.getChildren()) {
+            boolean found = false;
+            for(CaseStatement caze : cases) {
```
Comment (SpringVaS) [inline modules/fxgui/src/edu/kit/iti/algover/rule/script/ScriptController.java:409]: Name? Maye case?

**Comment by JonasKlamroth** [comment]
@SpringVaS i tried to fix the problems you pointed out. is this now working for you?

File: modules/fxgui/src/edu/kit/iti/algover/rule/script/ScriptController.java (hunk 1 above)
Comment (JonasKlamroth) [reply modules/fxgui/src/edu/kit/iti/algover/rule/script/ScriptController.java:407]: thats a jaava keyword and thus cant be used as a variable name :)

**Comment by SpringVaS** [comment]
General remark (not specific for this pull request): Case names cannot be altered in Script.

**Pull request merged**